pkg/pillar/zedUpload/awsutil/s3sync.go

```diff
@@ -29,6 +29,16 @@ type UpdateStats struct {
 
 type NotifChan chan UpdateStats
 
+// ChunkStats contians the details of Chunks being downloaded
+type ChunkStats struct {
+	Size  int64  // complete size to upload/download
+	Chunk []byte // chunk data being downloaded/uploaded
```

Comment:
Why do we need a copy of the data? That consumes 2x memory, and isn't part of any notion of stats I understand. Is the struct misnamed and should be called Chunk and not ChunkStats?

Reply:
> The time.Sleep really concerns me. Please explain.
> 
> Do we need to support SysOpDownloadByChunks for other datastores like http (with an implementation which doesn't chunk if the protocol doesn't support chunks)? If not, how does the users of this package know which datastores which methods?
> 
> We should have unittests for this but the tests can not contain credentials.
> 
> Names are confusing.
> 
> And is the code duplication necessary? Seems like the struct definiting and Writer are identical and can be shared.

time.Sleep -- > as and when the small chunk of data is download, I was writing into the prgChan channel and in-turn prgChan channel was post the data in req.result channel, the method was returning the moment we write on the prgChan, irrespective of whether channel finishes it's work.

SysOpDownloadByChunks --> supported for s3 and azure not sure about the other datastore

test case --> will running as part of zedCloud apiTest

Naming confusion --> fixing it

Duplicate code --> I didn't wanted to touch currently existing code/ struct, will fix the duplicate code issue.